### PR TITLE
Change bit operation to logical operation in TrackingTools/DetLayers

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This avoids clang warnings.

#### PR validation:

Compiling with clang no longer gives the warnings.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@makortel FYI

**Comment by Dr15Jones** [comment]
please test

File: TrackingTools/DetLayers/interface/rangesIntersect.h [hunk 1]
```diff
@@ -12,15 +12,15 @@
 
 template <typename Range>
 inline bool rangesIntersect(const Range& a, const Range& b) {
-  return !((a.first > b.second) | (b.first > a.second));
+  return !((a.first > b.second) || (b.first > a.second));
```
Comment (slava77) [inline TrackingTools/DetLayers/interface/rangesIntersect.h:15]: it looks like a wack-a-mole
https://github.com/cms-sw/cmssw/pull/40611#discussion_r1089028261

Comment (Dr15Jones) [reply TrackingTools/DetLayers/interface/rangesIntersect.h:15]: Is there anything to be done here?

**Comment by mandrenguyen** [comment]
+1

File: TrackingTools/DetLayers/interface/rangesIntersect.h (hunk 1 above)
Comment (mmusich) [reply TrackingTools/DetLayers/interface/rangesIntersect.h:15]: It looks like the bit operation here was intentional. Isn't there any way to instruct clang to not complain about these?

**Comment by mandrenguyen** [comment]
-reconstruction
pending response to Marco's comment above

**Comment by perrotta** [comment]
@Dr15Jones I think this can be closed, as it is going to be taken in charge by #40914

**Comment by perrotta** [comment]
Superseded by #40914

**Pull request closed without merging**